jupyter_client/channels.py

```diff
@@ -147,7 +144,10 @@ def run(self):
             # either a recv or connect, which cannot be followed by EFSM
             self.socket.send(b'ping')
             request_time = time.time()
-            ready = self._poll(request_time)
+            ready = False
+            while (not ready and not self._exit.is_set()
+                   and self.time_to_dead > (time.time() - request_time)):
+                ready = self._poll(.5)
```

Comment:
Is this the only usage of `_poll` in this repo?

Reply:
Yes

Reply:
Yes it seems so.